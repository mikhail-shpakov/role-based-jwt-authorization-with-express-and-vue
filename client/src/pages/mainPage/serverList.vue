<template lang="pug">
.container
  p.title.is-size-4 Список серверов
  p.subtitle
    | Нажмите на любой сервер, чтобы начать редактирование
    |  и Esc, чтобы выйти из режима редактирования
  .table-box
    b-table(
      :data="data"
      :columns="columns"
      :loading="isLoading"
      :selected.sync="selected"
      hoverable
      striped
    )
      template(slot='footer')
        .has-text-right Серверов всего: {{ data.length }}

      template(slot='empty')
          section.section
            .content.has-text-grey.has-text-centered
              p
                b-icon(
                  icon='emoticon-sad'
                  size='is-large'
                )
              p Список серверов пуст
</template>

<script>
export default {
  name: 'serverList',
  props: {
    data: {
      type: Array,
      required: true
    },
    isLoading: {
      type: Boolean,
      required: true
    }
  },
  data () {
    return {
      selected: {},
      columns: [
        {
          field: 'customer_id',
          label: 'ID',
          width: '60'
        },
        {
          field: 'server_name',
          label: 'Название сервера'
        },
        {
          field: 'server_type',
          label: 'Тип сервера'
        }
      ]
    }
  },
  methods: {
    addListenerEsc () {
      document.addEventListener('keyup', e => {
        if (e.code === 'Escape') this.selected = {}
      })
    }
  },
  watch: {
    selected () {
      this.$emit('selectServer', this.selected)
    }
  },
  mounted () {
    this.addListenerEsc()
  }
}
</script>

<style lang="sass" scoped>
div.table-box
  border: 1px solid $color-alt
  border-radius: 8px
  padding: 20px 10px 15px
  @media (min-width: $display-bp-mobile)
    padding: 22px 20px 20px

p.subtitle
  color: $color-alt
  font-size: 1rem
  margin: -1rem 0 2rem !important
</style>

<style lang="sass">
@media (max-width: $display-bp-mobile)
  div.b-table.has-mobile-cards tr
    box-shadow: none
    border-radius: 6px
    border: 1px solid $color-alt
    margin-top: 10px
    &.table-footer
      border: none
</style>
