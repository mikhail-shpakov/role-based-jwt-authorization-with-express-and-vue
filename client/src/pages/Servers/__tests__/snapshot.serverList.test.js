import { createLocalVue, shallowMount } from '@vue/test-utils'
import Buefy from 'buefy'
import serverList from '../serverList'

describe('./ServerList.vue', () => {
  const localVue = createLocalVue()
  localVue.use(Buefy)

  const data = [{
    user_id: 'user5',
    server_name: 'server2',
    server_type: 'dedicated'
  }]

  const isLoading = false

  it('when load page, list renders correctly', () => {
    const wrapper = shallowMount(serverList, {
      localVue,
      propsData: { data, isLoading }
    })
    expect(wrapper).toMatchSnapshot()
  })
})
