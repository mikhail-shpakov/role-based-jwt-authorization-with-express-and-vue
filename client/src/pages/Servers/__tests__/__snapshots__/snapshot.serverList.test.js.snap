// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./serverList.vue when load page, list renders correctly 1`] = `
<div class="container">
  <p class="title is-size-4">Список серверов</p>
  <p class="subtitle">Нажмите на любой сервер, чтобы начать редактирование
    и Esc, чтобы выйти из режима редактирования</p>
  <div class="table-box">
    <b-table-stub data="[object Object]" columns="[object Object],[object Object],[object Object]" striped="true" hoverable="true" headercheckable="true" checkboxposition="left" selected="[object Object]" isrowcheckable="function _default() {
        return true;
      }" checkedrows="" mobilecards="true" defaultsortdirection="asc" sorticon="arrow-up" sorticonsize="is-small" sortmultipledata="" currentpage="1" perpage="20" showdetailicon="true" paginationposition="bottom" rowclass="function _default() {
        return '';
      }" openeddetailed="" hasdetailedvisible="function _default() {
        return true;
      }" detailkey="" total="0"><template></template><template></template></b-table-stub>
  </div>
</div>
`;
